src/UnityExtension/Assets/Editor/GitHub.Unity/UI/PopupWindow.cs

```diff
@@ -0,0 +1,130 @@
+using System;
+using UnityEditor;
+using UnityEngine;
+
+namespace GitHub.Unity
+{
+    [Serializable]
+    class PopupWindow : BaseWindow
+    {
+        public enum PopupView
+        {
+            PublishView,
+            AuthenticationView
+        }
+
+        [MenuItem("GitHub/Authenticate")]
+        public static void Launch()
+        {
+            var popupWindow = (PopupWindow) Open(PopupView.AuthenticationView);
+            popupWindow.Initialize(EntryPoint.ApplicationManager);
+        }
+
+        [SerializeField] private PopupView activePopupView;
+        [SerializeField] private AuthenticationView authenticationView;
+        [SerializeField] private PublishView publishView;
+
+        [NonSerialized] private Subview activeSubview;
+
+        public Subview ActiveSubview
```

Comment:
A getter that sets values is confusing, and `ActiveSubview`/`ActivePopupView` isn't really clear about what is what. If we switch views by setting the enum value, then probably best to set the view there immediately.